tests/ssl-opt.sh

```diff
@@ -346,6 +346,56 @@ requires_openssl_legacy() {
     fi
 }
 
+requires_openssl() {
+    if [ -z "${OPENSSL_AVAILABLE:-}" ]; then
+        if which "${OPENSSL_CMD:-}" >/dev/null 2>&1; then
+            OPENSSL_AVAILABLE="YES"
+        else
+            OPENSSL_AVAILABLE="NO"
+        fi
+    fi
+    if [ "$OPENSSL_AVAILABLE" = "NO" ]; then
+        SKIP_NEXT="YES"
+    fi
+}
+# skip next test if tls1_3 is not available
+requires_openssl_tls1_3() {
+    requires_openssl
+    if [ "$OPENSSL_AVAILABLE" = "NO" ]; then
+        OPENSSL_TLS1_3_AVAILABLE="NO"
+    fi
+    if [ -z "${OPENSSL_TLS1_3_AVAILABLE:-}" ]; then
+        if $OPENSSL_CMD s_client -help 2>&1 | grep tls1_3 >/dev/null
```

Comment:
Hmm, the problem with that is the currently on the CI the default OpenSSL is 1.0.2g which doesn't have support for TLS 1.3, so the corresponding tests are always going to be skipped in the CI; we should use `$OPENSSL_NEXT` instead.

Apparently, this script is not using `OPENSSL_NEXT` yet, but you should be able to add support based on what's done for `GNUTLS_NEXT`and `OPENSSL_LEGACY`: set the variables `$O_NEXT_CLI` and `$O_NEXT_SRV` where similar variables are set for legacy, then change the previous function from `requires_openssl` to `requires_openssl_next` and using that here.